Examples/Heisenberg1d/heisenberg1d.py

```diff
@@ -0,0 +1,55 @@
+# Copyright 2021 The NetKet Authors - All rights reserved.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#    http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import netket as nk
+import numpy as np
+import jax
+
+# 1D Lattice
+L = 20
+
+g = nk.graph.Chain(length=L, pbc=True)
+# Lattice translation operations
+
+# Hilbert space of spins on the graph
+hi = nk.hilbert.Spin(s=1 / 2, N=g.n_nodes, total_sz=0)
+
+# Ising spin hamiltonian
+ha = nk.operator.Heisenberg(hilbert=hi, graph=g)
+
+# RBM Spin Machine
+ma = nk.models.create_RBMSymm(
```

Comment:
@femtobit I think I overlooked this, I thought we were still keeping ```RBMSymm`` as a simple constructor ?

Reply:
True, I was a bit confused by @PhilipVinc's request, though now I understand he only advocated to export both `DenseSymm` and `create_DenseSymm` for the layer and not for the RBM.

I can correct this - though treating layer and RBM differently still looks somewhat weird to me - while the layer is a bit more "internal", it is still part of the public API and definitely reusable outside of NetKet code.

Reply:
In the long run, we should probably find a solution along the lines of https://github.com/netket/netket/pull/573#issuecomment-804819668 (probably by discussing this with flax developers), which I could try in a few days.

Reply:
I agree 100% with your point that this is not consistent.

But my idea is that this is temporary until we find a decent workaround.
And it achieves the objective to keep a simple API for users that won't try to do more advanced things like defining their models.
In any case, if someone wants to define a model, I expect them to read at least a few bits of documentation (otherwise how can they know how to do that?), so they should see the warnings about `Create_DenseSymm`.